File: test_list_tools.mjs
import { spawn } from 'child_process';

const server = spawn('metabob-cli', ['mcp', '--transport', 'stdio'], {
  env: {
    ...process.env,
    METABOB_API_URL: 'http://localhost:8080',
    METABOB_PROJECT_ID: 'exp-repo-dev',
  },
  stdio: ['pipe', 'pipe', 'ignore']
});

let buffer = '';

server.stdout.on('data', (data) => {
  buffer += data.toString();
  const lines = buffer.split('\n');
  buffer = lines.pop() || '';
  
  lines.forEach(line => {
    if (line.trim()) {
      try {
        const msg = JSON.parse(line);
        if (msg.id === 1 && msg.result) {
          console.log('✓ Initialized');
          // Call listTools
          server.stdin.write(JSON.stringify({
            jsonrpc: '2.0', id: 2, method: 'tools/list', params: {}
          }) + '\n');
        }
        if (msg.id === 2 && msg.result) {
          console.log(`✓ Tools: ${msg.result.tools.length}`);
          const activityTools = msg.result.tools.filter(t => t.name.includes('activity') || t.name.includes('search'));
          console.log('\nActivity/Search tools:');
          activityTools.forEach(t => console.log(`  - ${t.name}`));
          server.kill();
          process.exit(0);
        }
      } catch (e) {}
    }
  });
});

server.stdin.write(JSON.stringify({
  jsonrpc: '2.0', id: 1, method: 'initialize',
  params: { protocolVersion: '2024-11-05', capabilities: {}, clientInfo: { name: 'test', version: '1.0' } }
}) + '\n');

setTimeout(() => { server.kill(); process.exit(1); }, 10000);

File: test_list_tools.py
#!/usr/bin/env python3
import subprocess
import json
import select
import sys
import os
import time

env = os.environ.copy()
env.update({
    'METABOB_API_URL': 'http://localhost:8080',
    'METABOB_API_KEY': 'mb_uYl7DfW-II6w-I9rR2-94iDKn1Hdu4QLm0wbFO9ePq8',
    'METABOB_PROJECT_ID': 'exp-repo-dev'
})

proc = subprocess.Popen(
    ['metabob-cli', 'mcp', '--transport', 'stdio'],
    stdin=subprocess.PIPE,
    stdout=subprocess.PIPE,
    stderr=subprocess.PIPE,
    env=env,
    text=True,
    bufsize=0
)

def send_message(msg):
    proc.stdin.write(json.dumps(msg) + '\n')
    proc.stdin.flush()

def wait_for_response(timeout=5):
    start = time.time()
    while time.time() - start < timeout:
        ready, _, _ = select.select([proc.stdout], [], [], 0.5)
        if ready:
            line = proc.stdout.readline()
            if line:
                return json.loads(line)
    return None

# Initialize
send_message({
    "jsonrpc": "2.0",
    "id": 1,
    "method": "initialize",
    "params": {
        "protocolVersion": "2024-11-05",
        "capabilities": {},
        "clientInfo": {"name": "test", "version": "1.0"}
    }
})
resp = wait_for_response()

# List tools
send_message({
    "jsonrpc": "2.0",
    "id": 2,
    "method": "tools/list"
})

resp = wait_for_response(timeout=10)
if resp:
    tools = resp.get("result", {}).get("tools", [])
    print(f"Found {len(tools)} tools:\n")
    for tool in tools:
        print(f"  - {tool['name']}")
        if 'activity' in tool['name'].lower() or 'search' in tool['name'].lower():
            print(f"    Description: {tool.get('description', 'N/A')}")
    print(f"\nActivity-related tools:")
    activity_tools = [t for t in tools if 'activity' in t['name'].lower()]
    for t in activity_tools:
        print(f"  {t['name']}: {t.get('description', '')[:80]}")
else:
    print("No response!")

proc.terminate()
